helix-term/src/commands.rs

```diff
@@ -5293,10 +5293,12 @@ fn select_register(cx: &mut Context) {
 }
 
 fn insert_register(cx: &mut Context) {
+    cx.editor.autoinfo = Some(Info::from_registers(&cx.editor.registers));
     cx.on_next_key(move |cx, event| {
         if let Some(ch) = event.char() {
-            cx.editor.selected_register = Some(ch);
-            paste_before(cx);
+            cx.editor.autoinfo = None;
+            cx.register = Some(ch);
+            paste_after(cx);
```

Comment:
`paste_before` matches kakoune behavior though.

Reply:
I think `insert_register` is a command used in insert mode. On Kakoune and Vi, `C-r` pastes a text at the cursor position, not before/after a selection range. Now I think `paste_after`, which I proposed, and `paste_before` are not the right solution. Wrong?

Reply:
Sorry, I'm wrong. 'paste_before` works perfectly.